superset/models/core.py

```diff
@@ -688,6 +691,12 @@ def _log_query(sql: str) -> None:
                     security_manager,
                 )
 
+        if self.db_engine_spec.engine == 'awsathena' and is_feature_enabled("DOWNLOAD_CSV_FROM_S3") and result_location == ChartDataResultLocation.S3:
```

Comment:
This too, should be moved to the DB engine spec so we don't have to explicitly test for Athena.

Something like:

```python
if self.db_engine_spec.supports_remote_download:
    url = self.db_engine_spec.get_remote_download_url(sqls[-1])
    ...
```